add LICENSE files to rtt_ros_integration packages

spd-intermodalics opened the pull request:
# Homogenize licenses in Orocos + ROS integration

This PR modifies the licenses of all the packages distributed in this repository to match the permissive BSD-3 clause license.
Previously, most of the packages already had a BSD license line in the `package.xml`. In some cases, this contradicted the header of some file.
So far, no author did actively choose the license, however, they were mostly remnants from other files as reference for headers.

After checking with all contributors to files that were licensed differently from BSD, we agreed to set this license for all the repository.

## Changes:

* A `LICENSE` file is given in the root of the repository with the contents of the BSD-3 clause license
* No authors / Copyright holders are modified from any file (source code that contained a header or `package.xml`)
* The licenses different to BSD-3 clause however are modified (source code that contained a header or `package.xml`)

spd-intermodalics committed:
add LICENSE files to rtt_ros_integration packages

spd-intermodalics committed:
rtt_tf: fix to the actual license of rtt_tf from source file

spd-intermodalics committed:
homogenize licenses to BSD

The patch reorganizes all the licesnes to match to BSD and a
single LICENSE file is placed in the root of the repository.

spd-intermodalics committed:
homogenize licenses to BSD ctd.

The patch reorganizes all the licesnes to match to BSD and a
single LICENSE file is placed in the root of the repository.

meyerj commented on rtt_ros/src/orocos/types/ros_float32_type.cpp:
```diff
@@ -1,38 +1,36 @@
-*       IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE    *
-*       POSSIBILITY OF SUCH DAMAGE.                                           *
-*                                                                             *
-*******************************************************************************/
+ *                 This file is part of the OROCOS toolchain ROS project       *
+ *                                                                             *
+ *        (C) 2010 Steven Bellens, steven.bellens@mech.kuleuven.be             *
+ *                    Department of Mechanical Engineering,                    *
+ *                   Katholieke Universiteit Leuven, Belgium.                  *
```
nit: Either we omit the trailing `*` for the whole copyright header, or add it as a box enclosing the whole license text with the correct width (without the additional space you introduced here)?

Suggestion:
```cpp
/*
 * This file is part of the OROCOS toolchain ROS project
 *
 * Copyright (C) 2010 Steven Bellens, steven.bellens@mech.kuleuven.be
 * Department of Mechanical Engineering,
 * Katholieke Universiteit Leuven, Belgium.
 *
 * ...
 */
```

following the example from [roscpp](https://github.com/ros/ros_comm/blob/kinetic-devel/clients/roscpp/include/ros/advertise_options.h), or

```cpp
// This file is part of the OROCOS toolchain ROS project
//
// Copyright (C) 2010 Steven Bellens, steven.bellens@mech.kuleuven.be
// Department of Mechanical Engineering,
// Katholieke Universiteit Leuven, Belgium.
//
// ...
```
which is the format applied in [rclcpp](https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/node.hpp) and many other ROS 2 packages, for example.

Either way is fine as long as it is somehow consistent across the whole repo.

meyerj commented on rtt_ros/src/orocos/types/ros_float32_type.cpp:
```diff
@@ -1,38 +1,36 @@
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **/
```
```suggestion
 */
```

No need for to repeat the `*` here and in other files.

meyerj reviewed (commented):
Until now, next to adding the top-level `LICENSE` file, this PR only modifies existing license headers in source files. But we should probably also add license headers in files that do not have any license header yet?

The copyright holder in source files should be consistent with the `<author>` information in `package.xml`, unless that is obviously wrong because of copy&paste. There can be multiple `<author>` tags, in case different files have different authors. Of course all the legal rights also apply to other contributors, even if they are not listed explicitly in the copyright statement, as mentioned in the license text itself.

Overall, the PR is a huge step towards fixing and clarifying the open source licenses, but it is also not yet complete and consistent for all files and packages. I mainly commented on minor style issues below.

Many thanks to all the original contributors who responded quickly and agreed to this license cleanup and partial re-licensing to BSD.

sportoles committed:
rtt_ros: fix license headers in types

This patch improves the layout of the BSD-3 clause licenses
in the package rtt_ros.

sportoles committed:
fix licenses format in source files

This patch applies a general homogeneous format for the licenses
accross the packages.

spd-intermodalics commented:
My view on when and how to modify source files was the next:
1) A general `LICENSE` file applies from the parent directory (root) or package `LICENSE` affecting to its tree.
2) When a license header existed in the source file, then I adapted it if it wasn't matching the tree license
3) When modifying a header license in a source file, I don't change the holder / owner / author, only the body of the license
4) When modifying a license with a license holder different from Intermodalics, I did it after collecting an explicit consent.
5) Style without modifying content may change in any file.

meyerj commented on LICENSE:
```diff
@@ -0,0 +1,26 @@
+Copyright (c) 2020, Intermodalics BVBA
```
```suggestion
Copyright (c) 2010, Katholieke Universiteit Leuven, Intermodalics BVBA and contributors
```

The year is typically the year of the first publication (e.g. from [here](https://en.wikipedia.org/wiki/Copyright_notice)). And it is probably better to list and acknowledge KU Leuven and other contributors, too, but without listing names of individuals? The authors of each contributions is preserved in the Git history.

meyerj commented on LICENSE:
```diff
@@ -0,0 +1,26 @@
+
+3. Neither the name of the copyright holder nor the names of its contributors
+may be used to endorse or promote products derived from this software without
+specific prior written permission.
```
formatting nit:
```suggestion
1. Redistributions of source code must retain the above copyright notice, this
   list of conditions and the following disclaimer.

2. Redistributions in binary form must reproduce the above copyright notice,
   this list of conditions and the following disclaimer in the documentation and/or
   other materials provided with the distribution.

3. Neither the name of the copyright holder nor the names of its contributors
   may be used to endorse or promote products derived from this software without
   specific prior written permission.
```

meyerj commented on rtt_ros/src/orocos/types/ros_float32_type.cpp:
```diff
@@ -1,38 +1,35 @@
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
```
Was this license text preserved from the old header ? It does not match the text of the [BSD-3-Clause license](https://opensource.org/licenses/BSD-3-Clause) in [`LICENSE`](https://github.com/orocos/rtt_ros_integration/pull/154/files?w=1#diff-9879d6db96fd29134fc802214163b95a), which must be copied without any changes.

This definitely needs to be fixed.

meyerj commented on typekits/rtt_kdl_conversions/package.xml:
```diff
@@ -4,7 +4,7 @@
   </description>
-  <license>GPL</license>
+  <license>BSD</license>
   <maintainer email="orocos-dev@lists.mech.kuleuven.be">Orocos Developers</maintainer>
```
Unrelated to licensing, but this email address does not exist anymore. I suggest to update it to `orocos-dev@orocos.org` in all packages, but we likely should designate a dedicated person or a smaller group of people in a follow-up PR.

meyerj commented on rtt_roscomm/include/rtt_roscomm/rostopic.h:
```diff
@@ -1,29 +1,34 @@
- *                                                                         *
- ***************************************************************************/
+/*
+ * tag: Ruben Smits  Tue Nov 16 09:18:49 CET 2010  rtt_rostopic.h
+ *
+ *        This file is part of the OROCOS toolchain ROS project
+ *
+ *         (C) 2010 Ruben Smits, ruben.smits@mech.kuleuven.be
```
Please remove the `tag:` line from all updated headers, not only the ones that have been modified in https://github.com/orocos/rtt_ros_integration/pull/154/commits/d620096ae4ab52e4ff93b1aab2dc65cd6d04bb4e. The information is redundant. In this particular case the file name is wrong anyway, because the header has been renamed a few years ago.

nit: Vertical whitespace is not consistent across different files, and I also propose to left-align all paragraphs after the asterisk boxes have been removed. The latter would make it easier to eventually update the copyright holder after the license header is copied to a new source or header file.

meyerj reviewed (changes requested):
The license text of the the [3-Clause BSD license](https://opensource.org/licenses/BSD-3-Clause) is wrong, or at least inconsistent. It has been partially preserved from the old headers, maybe an older version of the license text? There it was called the "Modified BSD License", but I also found other sources in the net with slightly different formulations.

Ideally all license headers in this repository should be identical in wording and format, apart from the copyright holder, after other licenses have been removed or changed with the consent of the contributors.

spd-intermodalics committed:
licenses: update text and make it match to current BSD-3

New update on the files with licenses headers.

spd-intermodalics committed:
update e-mail of Orocos Developers in package.xml files

meyerj reviewed (approved):
Thanks for updating!
